richtextfx-demos/src/main/java/org/fxmisc/richtext/demo/OverrideBehaviorDemo.java

```diff
@@ -0,0 +1,98 @@
+ * and temporarily via an InputMap. It also demonstrates the bugs that can arise if one adds a handler to
+ *  one of the `on[EventType]Property`. Rather than giving a guide on how to write InputMaptTemplates, there is
+ *  a link to StyledTextAreaBehavior as a list of examples to get one started.
+ */
+public class OverrideBehaviorDemo extends Application {
+
+    public static void main(String[] args) {
+        launch(args);
+    }
+
+    @Override
+    public void start(Stage primaryStage) {
+        InlineCssTextArea area = new InlineCssTextArea();
+
+        InputMap<Event> preventSelectionOrRightArrowNavigation = InputMap.consume(
+                anyOf(
+                        // prevent selection via (CTRL + ) SHIFT + [LEFT, UP, DOWN]
+                        keyPressed(LEFT,    SHIFT_DOWN, SHORTCUT_ANY),
+                        keyPressed(KP_LEFT, SHIFT_DOWN, SHORTCUT_ANY),
+                        keyPressed(UP,    SHIFT_DOWN, SHORTCUT_ANY),
+                        keyPressed(KP_UP, SHIFT_DOWN, SHORTCUT_ANY),
+                        keyPressed(DOWN,    SHIFT_DOWN, SHORTCUT_ANY),
+                        keyPressed(KP_DOWN, SHIFT_DOWN, SHORTCUT_ANY),
+
+                        // prevent selection via mouse events
+                        eventType(MouseEvent.MOUSE_DRAGGED),
+                        eventType(MouseEvent.DRAG_DETECTED),
+                        mousePressed().unless(e -> e.getClickCount() == 1 && !e.isShiftDown()),
+
+                        // prevent any right arrow movement, regardless of modifiers
+                        keyPressed(RIGHT,     SHORTCUT_ANY, SHIFT_ANY),
+                        keyPressed(KP_RIGHT,  SHORTCUT_ANY, SHIFT_ANY)
+                )
+        );
+        Nodes.addInputMap(area, preventSelectionOrRightArrowNavigation);
+
+        area.replaceText(String.join("\n",
+                "You can't move the caret to the right via the RIGHT arrow key in this area.",
+                "Additionally, you cannot select anything either",
+                "",
+                ":-p"
+        ));
+        area.moveTo(0);
+
+        CheckBox addExtraEnterHandlerCheckBox = new CheckBox("Temporarily add an EventHandler to area's `onKeyPressedProperty`?");
+        addExtraEnterHandlerCheckBox.setStyle("-fx-font-weight: bold;");
+
+        Label checkBoxExplanation = new Label(String.join("\n",
+                "The added handler will insert a newline character at the caret's position when [Enter] is pressed.",
+                "If checked, the default behavior and added handler will both occur: ",
+                "\tthus, two newline characters should be inserted when user presses [Enter].",
+                "When unchecked, the handler will be removed."
+        ));
+        checkBoxExplanation.setWrapText(true);
+
+        EventHandler<KeyEvent> insertNewlineChar = e -> {
+            if (e.getCode().equals(ENTER)) {
+                area.insertText(area.getCaretPosition(), "\n");
```

Comment:
You would want to call `e.consume()` here, to prevent the event from bubbling up the scene graph. The peculiarity is that other handlers on the same Node will still be invoked. 
From [Handling JavaFX Events](http://docs.oracle.com/javafx/2/events/processing.htm):

> However, if the node that consumes the event has more than one filter or handler registered for the event, the peer filters or handlers are still executed.